applications/minotari_ledger_wallet/wallet/src/handlers/get_script_signature.rs

```diff
@@ -3,98 +3,81 @@
 
 use alloc::format;
 
-use ledger_device_sdk::{io::Comm, ui::gadgets::SingleMessage};
-use tari_crypto::ristretto::{
-    pedersen::extended_commitment_factory::ExtendedPedersenCommitmentFactory,
-    RistrettoComAndPubSig,
-    RistrettoSecretKey,
+use blake2::Blake2b;
+use digest::consts::U64;
+use ledger_device_sdk::{io::Comm, random::Random, ui::gadgets::SingleMessage};
+use tari_crypto::{
+    commitment::HomomorphicCommitmentFactory,
+    keys::PublicKey,
+    ristretto::{
+        pedersen::{extended_commitment_factory::ExtendedPedersenCommitmentFactory, PedersenCommitment},
+        RistrettoComAndPubSig,
+        RistrettoPublicKey,
+        RistrettoSecretKey,
+    },
 };
 use zeroize::Zeroizing;
 
 use crate::{
     alloc::string::ToString,
+    hashing::{DomainSeparatedConsensusHasher, TransactionHashDomain},
     utils::{alpha_hasher, derive_from_bip32_key, get_key_from_canonical_bytes},
     AppSW,
     KeyType,
     RESPONSE_VERSION,
     STATIC_ALPHA_INDEX,
 };
 
-const MAX_TRANSACTION_LEN: usize = 312;
-pub struct ScriptSignatureCtx {
-    payload: [u8; MAX_TRANSACTION_LEN],
-    payload_len: usize,
-    account: u64,
-}
-
-// Implement constructor for TxInfo with default values
-impl ScriptSignatureCtx {
-    pub fn new() -> Self {
-        Self {
-            payload: [0u8; MAX_TRANSACTION_LEN],
-            payload_len: 0,
-            account: 0,
-        }
-    }
-
-    // Implement reset for TxInfo
-    fn reset(&mut self) {
-        self.payload = [0u8; MAX_TRANSACTION_LEN];
-        self.payload_len = 0;
-        self.account = 0;
-    }
-}
-
-pub fn handler_get_script_signature(
-    comm: &mut Comm,
-    chunk: u8,
-    more: bool,
-    signer_ctx: &mut ScriptSignatureCtx,
-) -> Result<(), AppSW> {
+pub fn handler_get_script_signature(comm: &mut Comm) -> Result<(), AppSW> {
     let data = comm.get_data().map_err(|_| AppSW::WrongApduLength)?;
 
-    if chunk == 0 {
-        // Reset transaction context
-        signer_ctx.reset();
-    }
-
-    if signer_ctx.payload_len + data.len() > MAX_TRANSACTION_LEN {
-        return Err(AppSW::ScriptSignatureFail);
-    }
-
-    // Append data to signer_ctx
-    signer_ctx.payload[signer_ctx.payload_len..signer_ctx.payload_len + data.len()].copy_from_slice(data);
-    signer_ctx.payload_len += data.len();
+    let mut account_bytes = [0u8; 8];
+    account_bytes.clone_from_slice(&data[0..8]);
+    let account = u64::from_le_bytes(account_bytes);
 
-    // If we expect more chunks, return
-    if more {
-        return Ok(());
-    }
+    let mut network_bytes = [0u8; 8];
+    network_bytes.clone_from_slice(&data[8..16]);
```

Comment:
Why not send 1 byte accross, as network is only 1 byte?

Reply:
There is a ticket to fix this but it's because there's only helpers for changing u64's to strings at the moment. If the value had to become a string format then I just used functions that existed, and kept them as u64's.